web/client/components/map/cesium/Map.jsx

```diff
@@ -497,15 +497,15 @@ class CesiumMap extends React.Component {
 
         const centerIsUpdate = !isNearlyEqual(newProps.center.x, currentCenter.longitude) ||
                                !isNearlyEqual(newProps.center.y, currentCenter.latitude);
-        const zoomChanged = newProps.zoom !== currentZoom;
+        const zoomChanged = Math.round(newProps.zoom) !== Math.round(currentZoom);      // round both to ignore the small tolerance due to calc zoom from hight
 
         // Do the change at the same time, to avoid glitches
         if (centerIsUpdate || zoomChanged) {
             const position = {
                 destination: Cesium.Cartesian3.fromDegrees(
-                    newProps.viewerOptions?.cameraPosition?.longitude ?? newProps.center.x,
-                    newProps.viewerOptions?.cameraPosition?.latitude ?? newProps.center.y,
-                    newProps.viewerOptions?.cameraPosition?.height ?? this.getHeightFromZoom(newProps.zoom ?? 0)
+                    newProps.mapStateSource !== this.props.id ? newProps?.center?.x ?? newProps.viewerOptions?.cameraPosition?.longitude : newProps.viewerOptions?.cameraPosition?.longitude ?? newProps.center.x,
+                    newProps.mapStateSource !== this.props.id ? newProps?.center?.y ?? newProps.viewerOptions?.cameraPosition?.latitude : newProps.viewerOptions?.cameraPosition?.latitude ?? newProps.center.y,
+                    newProps.mapStateSource !== this.props.id ? this.getHeightFromZoom(newProps.zoom ?? 0) ?? newProps.viewerOptions?.cameraPosition?.height : newProps.viewerOptions?.cameraPosition?.height ?? this.getHeightFromZoom(newProps.zoom ?? 0)
```

Comment:
Please restore this to the previous arguments, if we want to propertly control the camera direction we will need to include the [viewerOptions](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/components/map/enhancers/onMapViewChanges.js#L17) property to the widgets dependecies.